### PR TITLE
[ICTL-893] removing TestsExecutionResultService

### PR DESCRIPTION
**Pull request by NikolaiSviridov**
# Description of changes made
* Removed TestsExecutionResultService
* Refactoring TestCasePanelFacrory -> TestCasePanel

# Why is merge request needed
Issue ICTL-893

*Please write if you have anything to add*

TestCasePanelFacrory wasn't a factory from the beginning. Also, it needs to be refactored again, to contain checkbox logic inside, instead of using JPanels outside the object. 

- [x] I have checked that I am merging into correct branch

### CODE REVIEW

**Review by arksap2002 — APPROVED** [review]

**Review by arksap2002 — APPROVED** [review]

**Review by pderakhshanfar — APPROVED** [review]

File: src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt [hunk 1]
```diff
@@ -95,6 +95,10 @@ class TestCasePanelBuilder(
 
     private var isRemoved = false
 
+    private var _error: String? = null
+    val error: String?
+        get() = _error
+
```
Comment (Vladislav0Art) [inline src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt:101]: Would not the following be the same?
```kotlin
var error: String? = null
  private set
```

File: src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt [hunk 2]
```diff
@@ -136,6 +140,14 @@ class TestCasePanelBuilder(
...
+    init {
+        prepareUpperPanel()
+        prepareMiddlePanel()
+        prepareBottomPanel()
+        updateErrorRelatedUI()
+    }
 
     /**
      * Retrieves the upper panel for the GUI.
```
Comment (Vladislav0Art) [inline src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt:153]: `Retrieves` -> `builds/assembles`, as the implementation does not retrieve the panel from anywhere, but populates it with UI elements.

Comment (Vladislav0Art) [reply src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt:153]: Same for the two other analogous methods

File: src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt [hunk 3]
```diff
@@ -344,6 +336,18 @@ class TestCasePanelBuilder(
         })
     }
 
+    private fun updateErrorRelatedUI() {
+        updateBorder(error)
+        if (error.isNullOrBlank()) {
+            errorLabel.isVisible = false
+        } else {
+            errorLabel.isVisible = true
+            errorLabel.toolTipText = error
```
Comment (Vladislav0Art) [inline src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt:345]: I do not know how the render process works for IntelliJ plugins, but if it is reactive, then here exists a moment when the `errorLabel` is visible but yet contains no error message. In this case, it is advisable to swap the assignments to:
```kotlin
errorLabel.toolTipText = error
errorLabel.isVisible = true
```

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTestsTab/GeneratedTestsTabBuilder.kt [hunk 4]
```diff
@@ -209,7 +207,7 @@ class GeneratedTestsTabBuilder(
     private fun applyTests() {
         // Filter the selected test cases
         val selectedTestCasePanels =
-            generatedTestsTabData.testCasePanels.filter { (it.value.getComponent(0) as JCheckBox).isSelected }
+            generatedTestsTabData.testCaseNameToPanels.filter { (it.value.getComponent(0) as JCheckBox).isSelected }
```
Comment (Vladislav0Art) [inline src/main/kotlin/org/jetbrains/research/testspark/display/generatedTestsTab/GeneratedTestsTabBuilder.kt:210]: What component resides under the 0th index in `getComponent(0)`? It is better to comment on it or, even better, move the `0` into a named constant.

File: src/main/kotlin/org/jetbrains/research/testspark/uiUtils/GenerateTestsTabHelper.kt [hunk 5]
```diff
@@ -14,7 +14,7 @@ object GenerateTestsTabHelper {
      * @return the editor corresponding to the test case, or null if it does not exist
      */
     fun getEditorTextField(testCaseName: String, generatedTestsTabData: GeneratedTestsTabData): EditorTextField? {
-        val middlePanelComponent = generatedTestsTabData.testCasePanels[testCaseName]?.getComponent(2) ?: return null
+        val middlePanelComponent = generatedTestsTabData.testCaseNameToPanels[testCaseName]?.getComponent(2) ?: return null
```
Comment (Vladislav0Art) [inline src/main/kotlin/org/jetbrains/research/testspark/uiUtils/GenerateTestsTabHelper.kt:17]: What resides under `getComponent(2)`? it is better to comment on it or move into a named constant.

**Review by Vladislav0Art — CHANGES_REQUESTED** [review]

File: src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt (hunk 2 above)
Comment (NikolaiSviridov) [reply src/main/kotlin/org/jetbrains/research/testspark/display/TestCasePanel.kt:152]: it's an old explanation, I'll change it

File: src/main/kotlin/org/jetbrains/research/testspark/display/generatedTestsTab/GeneratedTestsTabBuilder.kt (hunk 4 above)
Comment (NikolaiSviridov) [reply src/main/kotlin/org/jetbrains/research/testspark/display/generatedTestsTab/GeneratedTestsTabBuilder.kt:211]: @arksap2002 please check this out

File: src/main/kotlin/org/jetbrains/research/testspark/uiUtils/GenerateTestsTabHelper.kt (hunk 5 above)
Comment (NikolaiSviridov) [reply src/main/kotlin/org/jetbrains/research/testspark/uiUtils/GenerateTestsTabHelper.kt:17]: @arksap2002

**Pull request merged**